src/octree/octree_iterator.rs

```diff
@@ -130,3 +130,47 @@ pub fn intersecting_node_ids(
     }
     node_ids
 }
+
+pub struct NodeIdsIterator<'a, F>
+where
+    F: Fn(&NodeId, &Octree) -> bool + 'a,
+{
+    octree: &'a Octree,
+    filter_func: F,
+    node_ids: VecDeque<NodeId>,
+}
+
+impl<'a, F> NodeIdsIterator<'a, F>
+where
+    F: Fn(&NodeId, &Octree) -> bool + 'a,
+{
+    pub fn new(octree: &'a Octree, filter_func: F) -> Self {
+        NodeIdsIterator {
+            octree,
+            node_ids: vec![NodeId::from_level_index(0, 0)].into(),
+            filter_func,
+        }
+    }
+}
+
+impl<'a, F> Iterator for NodeIdsIterator<'a, F>
+where
+    F: Fn(&NodeId, &Octree) -> bool + 'a,
```

Comment:
I don't think we need that here when it's already in the struct definition, do we?

Reply:
Same above

Reply:
Unfortunately it does not work removing that